adoc/chapters/programming_interface.adoc

```diff
@@ -22074,10 +22077,12 @@ used for forward compatibility with future SYCL versions.
 include::{header_dir}/algorithms/reduce.h[lines=4..-1]
 ----
 
-  . _Constraints:_ Available only if
-    [code]#sycl::is_group_v<std::decay_t<Group>># is true, [code]#Ptr# is a
-    pointer to a fundamental type, and [code]#BinaryOperation# is a SYCL
-    function object type.
+  . _Constraints:_ Available only if all of the following conditions are met:
+    - [code]#sycl::is_group_v<std::decay_t<Group>># is true;
+    - [code]#Ptr# is a pointer or [code]#multi_ptr# to: a fundamental type,
+      [code]#sycl::half#, [code]#marray# or [code]#vec#; and
```

Comment:
Note that `marray` is allowed to have elements that are not fundamental types.  In fact, the elements are even allowed to be types that are not trivially copyable (or even device copyable).  I suspect you do not intend to support `marray` in these cases?

Reply:
Good catch.  Fixed in cd2801cd02b3cd8d7c8261099e46eeecd3223eb7.

I used the wording "`marray` of a fundamental type", because there are quite a few places in the specification where we already use "`marray` of `T`" to mean "an instantiation of `marray` where `DataElement` is `T`".  For example, the descriptions of the various math overloads often say "pointer to `marray` of `int`".